injector/cpu_pressure.go

```diff
@@ -7,155 +7,91 @@ package injector
 
 import (
 	"fmt"
-	"runtime"
-	"sync"
+	"math"
 
 	"github.com/DataDog/chaos-controller/api/v1beta1"
 	"github.com/DataDog/chaos-controller/process"
-	"github.com/DataDog/chaos-controller/stress"
 	"github.com/DataDog/chaos-controller/types"
+	"k8s.io/apimachinery/pkg/util/intstr"
 )
 
-type cpuPressureInjector struct {
-	spec     v1beta1.CPUPressureSpec
-	config   CPUPressureInjectorConfig
-	routines int
+type CPUStressArgsBuilder interface {
+	GenerateArgs(int) []string
 }
 
-// CPUPressureInjectorConfig is the CPU pressure injector config
-type CPUPressureInjectorConfig struct {
-	Config
-	Stresser        stress.Stresser
-	StresserExit    chan struct{}
-	ProcessManager  process.Manager
-	StresserManager stress.StresserManager
+type cpuPressureInjector struct {
+	config                   Config
+	spec                     *v1beta1.CPUPressureSpec
+	backgroundProcessManager process.BackgroundProcessManager
+	backgroundProcess        process.BackgroundProcess
+	cpuStressArgsBuilder     CPUStressArgsBuilder
 }
 
 // NewCPUPressureInjector creates a CPU pressure injector with the given config
-func NewCPUPressureInjector(spec v1beta1.CPUPressureSpec, config CPUPressureInjectorConfig) (Injector, error) {
-	// create stresser
-	if config.Stresser == nil {
-		config.Stresser = stress.NewCPU(config.Disruption.DryRun)
-	}
-
-	if config.StresserExit == nil {
-		config.StresserExit = make(chan struct{})
-	}
-
-	// create process manager
-	if config.ProcessManager == nil {
-		config.ProcessManager = process.NewManager(config.Disruption.DryRun)
-	}
-
-	if config.StresserManager == nil {
-		return nil, fmt.Errorf("StresserManager does not exist")
-	}
-
+func NewCPUPressureInjector(config Config, spec *v1beta1.CPUPressureSpec, backgroundProcessManager process.BackgroundProcessManager, argsBuilder CPUStressArgsBuilder) Injector {
 	return &cpuPressureInjector{
-		spec:   spec,
-		config: config,
-	}, nil
+		config:                   config,
+		spec:                     spec,
+		backgroundProcessManager: backgroundProcessManager,
+		cpuStressArgsBuilder:     argsBuilder,
+	}
 }
 
 func (i *cpuPressureInjector) GetDisruptionKind() types.DisruptionKindName {
 	return types.DisruptionKindCPUPressure
 }
 
 func (i *cpuPressureInjector) Inject() error {
-	// move the injector process to the target cgroup controllers
-	i.config.Log.Infow("joining target cgroups")
```

Comment:
nit: can we keep this debug log? It could help to insure the Inject function has been called.